gunpowder/nodes/upsample.py

```diff
@@ -80,20 +83,25 @@ def process(self, batch, request):
         input_roi = batch.arrays[self.source].spec.roi
         request_roi = request[self.target].roi
 
-        assert input_roi.contains(request_roi)
-
-        # upsample
-
-        logger.debug("upsampling %s with %s", self.source, self.factor)
+        # get roi expanded to source voxel size
+        source_voxel_size = self.spec[self.source].voxel_size
+        expanded_request_roi = request_roi.snap_to_grid(source_voxel_size, mode='grow')
+        logger.debug("expanded request ROI is %s"%request_roi)
+        assert input_roi.contains(expanded_request_roi)
```

Comment:
Not sure if this assert is needed either. `request_roi.snap_to_grid(source_voxel_size, mode='grow')` is the same thing that is done in the prepare function, so it is safe to assume that this is what you receive in the batch. I think nodes should avoid asserting that other components of gunpowder work as expected.